### DISCUSSION TITLE
New display feature: wiring?

### DISCUSSION THREAD

**Discussion by casimirextreme** [tbnobody/OpenDTU#532] [General]
I've a SSD1306 screen and would like to add it to my current ESP32 + NRF2401 setup

Is there any wiring schema that could be shared like the one on the README?

Sorry to ask this question but still learning and not familiar with electronics. The main question I have is that the NRF2401 is already connected to the 3.3v pin and ground. May I connect both the SSD1306 and the NRF2401 on the same respective pins for 3.3v and GND?

Thanks a lot.

**Comment by Archibald-Harry-Tuttle** [comment]
ESP32 what? -32U ; poe ;-c3 ;-s2 ; dev-Board XY
30 or 38pin?

First:
Yes, you use both SSD1306 and NRF24l01+ on same +3.3V and GND
My ESP32 Wroom 32 -32U /38pin has
SDA from Display on GIO 21 and
SDK from Display on GIO 22

**Reply by laurent-luyckx-keemotion** [reply]
Yes I'm using the same ESP32 as on the README (38pins).
Thanks for you reply, it replies to my main question (sharing the same pins for 3.3v and GND)

**Comment by tbnobody** [comment]
I've added a first small documentation [here](https://github.com/tbnobody/OpenDTU/blob/master/docs/Display.md)

**Reply by laurent-luyckx-keemotion** [reply]
Thank you so much!

**Comment by MichaMEG** [comment]
Vielen Dank, sieht gut aus, bin gespannt, ob es klappt. 
Ich hoffe mein Display kommt morgen.

**Reply by MichaMEG** [reply]
Great, it works. :+1: 
What irritated me at first was that the display showed nothing because the inverter is currently not producing anything.
After switching off Powersave it worked.

Am curious how it looks tomorrow in daylight.

**Comment by SteffMUC** [comment]
Hi, 
Just jumping into this thread, since you have been successful to get it working with a **SSD1306**. 
I have hooked the SSD1306 to a ESP32 NodeMCU Development Board. 
OpenDTU is running smoothly since quite a while. 
The display is connected to 
```
SDA from Display on GIO 21 and 
SDK from Display on GIO 22.
``` 

Did measure the voltage, yes it is supplied with 3.3.V.
I did you use config file "nodemcu_esp32.json" with no changes from this repository. Within the GUI I did select "Generic NodeMCU 32 with SSD1306". I turned off powersafe, I waited until we got sun, but still nothing is displayed on the SSD1306.

Initially I thought the display might be defective, so I tested it yesterday with a small python script on a raspi: Yes it is working there.

I tested the connection wires for SDA and SDK and they are connected to the proper pins.

Any idea/ hint what still could be potentially wrong ?

Thanks for any input...

Latest FW version [f87f265](https://github.com/tbnobody/OpenDTU/commits/f87f265)

Potentially something very simple with my first self-made PCB, but I cannot find the mistake...:
https://github.com/SteffMUC/openDTU_wDisplay

**Reply by SteffMUC** [reply]
Solved it myself, change of the FW helped tremendously, now it is running with FW
https://github.com/tbnobody/OpenDTU/commits/b5d099f
![IMG_0541](https://user-images.githubusercontent.com/25563500/218255701-ce73c82c-d3aa-4174-81fc-979646e7f8ef.jpg)

**Reply by SteffMUC** [reply]
In case someone is interested/ want to replicate, made a small PCB for directly mounting the display and additional case for it as well, working now reliably since a few days: https://github.com/SteffMUC/openDTU_wDisplay2

**Comment by huckes** [comment]
Hi, I compiled the latest code for a Wemos D1 mini esp32 by changing  the platformio_override.ini with

[platformio]
default_envs = d1 mini esp32

and for the Pin Mapping I defined a Wemos_ESP32_D1_Mini.json with/without SSD1306-display with this content:

[
    {
        "name": "Wemos ESP32 D1 Mini",
        "nrf24": {
            "miso": 19,
            "mosi": 23,
            "clk": 18,
            "irq": 16,
            "en": 17,
            "cs": 5
        },
        "eth": {
            "enabled": false,
            "phy_addr": -1,
            "power": -1,
            "mdc": -1,
            "mdio": -1,
            "type": 0,
            "clk_mode": 0
        }
    },
    {
        "name": "Wemos ESP32 D1 Mini with SSD1306",
        "nrf24": {
            "miso": 19,
            "mosi": 23,
            "clk": 18,
            "irq": 16,
            "en": 17,
            "cs": 5
        },
        "eth": {
            "enabled": false,
            "phy_addr": -1,
            "power": -1,
            "mdc": -1,
            "mdio": -1,
            "type": 0,
            "clk_mode": 0
        },
        "display": {
            "type": 2,
            "data": 21,
            "clk": 22
        }
    }
]

**Comment by mischerr** [comment]
I've seen that there is an integration for the PCD8544 (Nokia 5110).
Is there an instruction / schemantic on how to connect?

**Comment by beneva52** [comment]
[https://github.com/tbnobody/OpenDTU/discussions/591](url)
